native/src/main/java/io/ballerina/persist/cmd/Generate.java

```diff
@@ -0,0 +1,252 @@
+ */
+@CommandLine.Command(
+        name = "generate",
+        description = "generate database configurations."
+        )
+
+public class Generate implements BLauncherCmd {
+
+    private static final PrintStream errStream = System.err;
+
+    private String sourcePath = "";
+    public ProjectEnvironmentBuilder projectEnvironmentBuilder;
+
+    private static final String COMMAND_IDENTIFIER = "persist-generate";
+
+    private CommandLine parentCmdParser;
+
+    Project balProject;
+
+    String name;
+
+    @CommandLine.Option(names = {"-h", "--help"}, hidden = true)
+    private boolean helpFlag;
+
+    @Override
+    public void execute() {
+        if (helpFlag) {
+            String commandUsageInfo = BLauncherCmd.getCommandUsageInfo(COMMAND_IDENTIFIER);
+            errStream.println(commandUsageInfo);
+            return;
+        }
+        try  {
+            if (projectEnvironmentBuilder == null) {
+                balProject = ProjectLoader.loadProject(Paths.get(""));
+            } else {
+                balProject = ProjectLoader.loadProject(Paths.get(this.sourcePath), projectEnvironmentBuilder);
+            }
+            name = balProject.currentPackage().descriptor().org().value() + "." + balProject.currentPackage()
+                    .descriptor().name().value();
+
+
+        } catch (ProjectException e) {
+            errStream.println("The current directory is not a Ballerina project!");
+            return;
+        }
+        try {
+            ArrayList<Entity> entityArray = readBalFiles();
+            if (entityArray.size() == 0) {
+                errStream.println("Current directory doesn't contain any persist entities");
+            } else {
+                for (Entity entity : entityArray) {
+                    entity.setPackageName(balProject.currentPackage().descriptor().org().value() + "/"
+                            + balProject.currentPackage().descriptor().name().value());
+                    generateClientBalFile(entity);
+                }
+                generateConfigurationBalFile();
+            }
+        } catch (Exception e) {
+            errStream.println(e.getMessage());
+        }
+    }
+
+    private ArrayList<Entity> readBalFiles() throws BalException {
+        ArrayList<Entity> returnMetaData = new ArrayList<>();
+        Path dirPath = Paths.get(this.sourcePath);
+        List<Path> fileList;
+
+
+        try (Stream<Path> walk = Files.walk(dirPath)) {
+            if (walk != null) {
+                fileList = walk.filter(Files::isRegularFile).collect(Collectors.toList());
+                if (hasSyntacticDiagnostics(Paths.get(this.sourcePath)) ||
+                        hasSemanticDiagnostics(Paths.get(this.sourcePath), this.projectEnvironmentBuilder)) {
+                    throw new BalException("Error occurred while building the project!");
+                }
+                for (Path filePath : fileList) {
+                    if (filePath.toString().endsWith(".bal")) {
```

Comment:
can we use `Files.getFileExtension(path);` and compare

Reply:
is this a part of `java.nio.file.Files`? I cannot find this method in the [documentation] (https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html)

Reply:
No, it is from the guava library. we can't use it.